addons/mail/static/src/core/common/store_service.js

```diff
@@ -245,16 +211,17 @@ export const storeService = {
                                 }
                                 return Reflect.get(target, name, receiver);
                             },
-                            deleteProperty(target, key) {
+                            deleteProperty(target, name) {
                                 if (name !== "__rels__" && target.__rels__.has(name)) {
                                     const r1 = target;
                                     const l1 = r1.__rels__.get(name);
                                     const r2 = res.get(l1);
                                     if (r2) {
                                         r2.__invs__.delete(r1.localId, name);
                                     }
+                                    r1.__rels__.set(name, undefined);
                                 }
-                                const ret = Reflect.deleteProperty(target, key);
+                                const ret = Reflect.deleteProperty(target, name);
```

Comment:
The code was wrong in `deleteProperty`. With introduction of `DiscussApp` and `DiscussAppCategory` models, some tests show the problems, hence the fix in the diff